scripts/airdropBsxToHdx/getVestingSchedules.ts

```diff
@@ -0,0 +1,121 @@
+// ✅  -> calculate BSX reward for accounts from .json of getData
+
+// ✅ -> calulate 5 000 000 000 proportional to holdings at the time for each holder
+//  ->  output a .json file of all schedules
+
+// ✅  -> start vesting at 9 334 719, Crowd Loan BSX vest
+// ✅  -> end vesting ~22 months after start
+
+const bignumber = require("bignumber.js")
+const airdropData = require("./data/airdropData.json")
+
+type DynamicVestingInfo = {
+    destination: string,
+    schedule: {
+        amountToBeVested: string,
+        start: string,
+        period: string,
+        per_period: string,
+        period_count: string 
+    }
+}
+
+type VestingBatch = DynamicVestingInfo[]
+
+const tenTo12Power = new bignumber('10').pow('12');
+
+const airdropBsxAllocation = 
+    new bignumber('100000000000') // 100b -> total supply of BSX
+    .multipliedBy(new bignumber('0.05'))
+    .multipliedBy(tenTo12Power).toFixed(0)
+// ^^^ 5% of total supply
+
+console.log(
+    '\n total allocation to non-investor, non-founder accounts: ', 
+    new bignumber(airdropBsxAllocation).div(tenTo12Power).toString() + '\n'
+)
+
+const sumHdxBalances = new bignumber(airdropData.totalAllOgHdxBalances)
+
+const bsxAirdroppedPerOneHdx = 
+    new bignumber(airdropBsxAllocation).div(sumHdxBalances)
+
+console.log(
+    '\n amount of BSX tokens you get per one HDX token: ', 
+    bsxAirdroppedPerOneHdx.toString() + '\n'
+)
+
+const leaseStartBlock = '9334719'
+    // the auction we won ended on this block 
+    // ^^^ _9 334 719_ https://kusama.subscan.io/auction/8
+
+const twentyTwoMonthsInBlocks = (
+    60/    // seconds in minute
+    6.5*   // seconds to produce a block -> https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/explorer
```

Comment:
It's 12 seconds on [Basilisk](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fbasilisk.api.onfinality.io%2Fpublic-ws#/explorer). And I would place the division at the end.

Reply:
vesting is based on the relay chain blocks, that are more reliable every 6 secs